### ISSUE TITLE
[Research] OUI Compliance audit of the `discover` plugin

### ISSUE THREAD

**Issue by BSFishy** [opensearch-project/OpenSearch-Dashboards#4079]
Tracking issue for OUI compliance audit of the `discover` plugin.

Follow the steps in #3945 to complete the audit, then open another issue with findings. Once the work of performing the audit is done, this issue can be closed.

<details>
  <summary>Related files</summary>

  size (bytes) | file | notes
---|---|---
2.7K | [src/plugins/discover/public/application/_discover.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/_discover.scss) | 
2.6K | [src/plugins/discover/public/application/angular/doc_table/_doc_table.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/doc_table/_doc_table.scss) | 
1.9K | [src/plugins/discover/public/application/components/sidebar/discover_sidebar.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/components/sidebar/discover_sidebar.scss) | 
1.1K | [src/plugins/discover/public/application/components/doc_viewer/doc_viewer.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/components/doc_viewer/doc_viewer.scss) | 
888 | [src/plugins/discover/public/application/angular/doc_table/components/table_row/_open.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/doc_table/components/table_row/_open.scss) | 
670 | [src/plugins/discover/public/application/angular/doc_table/components/table_row/_cell.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/doc_table/components/table_row/_cell.scss) | 
520 | [src/plugins/discover/public/application/angular/doc_table/components/table_row/_details.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/doc_table/components/table_row/_details.scss) | 
281 | [src/plugins/discover/public/application/embeddable/search_embeddable.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/embeddable/search_embeddable.scss) | 
236 | [src/plugins/discover/public/application/angular/context/components/action_bar/_action_bar.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/context/components/action_bar/_action_bar.scss) | 
227 | [src/plugins/discover/public/application/angular/doc_table/components/_table_header.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/doc_table/components/_table_header.scss) | 
201 | [src/plugins/discover/public/application/angular/context/_index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/context/_index.scss) | 
132 | [src/plugins/discover/public/application/components/sidebar/discover_field_bucket.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/components/sidebar/discover_field_bucket.scss) | 
121 | [src/plugins/discover/public/application/components/sidebar/discover_field_details.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/components/sidebar/discover_field_details.scss) | 
101 | [src/plugins/discover/public/application/angular/directives/_histogram.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/directives/_histogram.scss) | 
79 | [src/plugins/discover/public/application/components/sidebar/discover_field.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/components/sidebar/discover_field.scss) | 
53 | [src/plugins/discover/public/application/angular/_index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/_index.scss) | 
51 | [src/plugins/discover/public/application/angular/doc_table/components/table_row/_index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/doc_table/components/table_row/_index.scss) | 
51 | [src/plugins/discover/public/application/angular/doc_table/components/_index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/doc_table/components/_index.scss) | 
49 | [src/plugins/discover/public/application/angular/doc_table/index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/doc_table/index.scss) | 
45 | [src/plugins/discover/public/application/index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/index.scss) | 
43 | [src/plugins/discover/public/application/angular/directives/_index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/directives/_index.scss) | 
39 | [src/plugins/discover/public/application/angular/directives/_no_results.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/directives/_no_results.scss) | 
22 | [src/plugins/discover/public/application/angular/context/components/action_bar/_index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/discover/public/application/angular/context/components/action_bar/_index.scss) | 

</details>

If this issue isn't already assigned to someone and you'd like to take on this work, please ping @BSFishy in the comments and I can assign it to you.

**Comment by BSFishy** [comment]
@ashwin-pc do we need this audit, given Discover 2.0?

**Comment by BSFishy** [comment]
We can skip this audit, as this work will be done when we do Discover 2.0

**Comment by joshuarrrr** [comment]
Reopening because I do think we'll need to do this after https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4216 or incorporate it there.
